### PR TITLE
[STAKAN-48] Stake dashboard

### PR DESCRIPTION
**Pull request by BigBadAlien**
Hello, brave developer, nice day to merge a pull request, ain't it?

## Follow the checklist to proceed

- [x] I've performed a code self-review
- [x] I've tested feature or bugfix on my own
- [x] If I make changes to another person's module, I've asked how to use it or request a review
- [x] I've checked mobile & tablet screens (if have changes in UI)
- [x] Local build successfully completed
- [x] Branch name and pull request title are according to [the requirements](https://ankrnetwork.atlassian.net/wiki/spaces/STAKAN/pages/1666646162/Branches+and+pull+request+naming)

### CODE REVIEW

File: src/modules/stake-dot/StakeDotRoutes.tsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+export enum ParachainNetwork {
+  DOT = 'DOT',
+  KSM = 'KSM',
+  WND = 'WND',
+  ROC = 'ROC',
+}
+
+export const StakeDotRoutesConfig = {
+  dashboard: {
```
Comment (Antroll) [inline src/modules/stake-dot/StakeDotRoutes.tsx:19]: Maybe we should create an `IRoutesConfig` interface?

File: src/modules/stake-dot/components/Claim/ClaimStyles.tsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+      },
+      amount: {
+        display: 'grid',
+        gridTemplateColumns: 'auto auto',
+        paddingRight: theme.spacing(2),
+        gridGap: theme.spacing(1),
+        alignItems: 'baseline',
+      },
+      amountLabel: {
+        fontSize: props => 38 - 8 * (props.amount.toFixed().length - 2),
```
Comment (Antroll) [inline src/modules/stake-dot/components/Claim/ClaimStyles.tsx:48]: As it seems to me, this is an unnecessary complication of styling

File: src/modules/stake-dot/components/Dashboard/DashboardStyles.tsx [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+export const useDashboardStyles = makeStyles<Theme>(theme => ({
+  root: {
+    marginTop: theme.spacing(10),
+  },
+  header: {
+    display: 'flex',
+    flexDirection: 'row',
+    justifyContent: 'start',
+    margin: `${theme.spacing(6)}px auto ${theme.spacing(3)}px`,
```
Comment (Antroll) [inline src/modules/stake-dot/components/Dashboard/DashboardStyles.tsx:11]:
```suggestion
    margin: theme.spacing(6, 'auto', 3),
```

File: src/modules/stake-dot/components/Dashboard/DashboardStyles.tsx [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    [theme.breakpoints.down('lg')]: {
+      padding: theme.spacing(0, 10),
+    },
+  },
+  stats: {
+    display: 'grid',
+    gridTemplateColumns: '1fr 1fr',
+    gridTemplateRows: '1fr',
+    gap: theme.spacing(4, 4),
+    margin: `${theme.spacing(6)}px auto ${theme.spacing(3)}px`,
```
Comment (Antroll) [inline src/modules/stake-dot/components/Dashboard/DashboardStyles.tsx:24]:
```suggestion
    margin: theme.spacing(6, 'auto', 3),
```

File: src/modules/stake-dot/components/StakeDotAlert/StakeDotAlert.tsx [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+export const StakeDotAlert = () => {
+  const classes = useStakeDotAlertStyles();
+
+  useInitEffect(() => {
+    const interval = setInterval(() => {
+      console.log('dispatch check wallet');
+    }, REFRESH_INTERVAL);
+
+    return () => clearInterval(interval);
+  });
```
Comment (Antroll) [inline src/modules/stake-dot/components/StakeDotAlert/StakeDotAlert.tsx:20]: It seems we have a [hook](https://github.com/Ankr-network/stkr-web3/blob/stage/src/common/utils/useInterval.ts) for this.

File: src/modules/stake-dot/screens/StakeDot/StakeDot.tsx [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+  useInitEffect(() => {
+    dispatch(AvalancheActions.fetchTransactionStatus());
+  });
+
+  const renderAlert = useCallback(() => <StakeDotAlert />, []);
+
+  if (needAlert) {
+    // TODO: check connection and requiredNetwork
+    return renderAlert();
+  }
```
Comment (Antroll) [inline src/modules/stake-dot/screens/StakeDot/StakeDot.tsx:22]: I don't understand why we have to use an additional callback to render the alert?

```suggestion
  if (needAlert) {
    // TODO: check connection and requiredNetwork
    return <StakeDotAlert />;
  }
```

File: src/modules/stake-dot/StakeDotRoutes.tsx (hunk 1 above)
Comment (BigBadAlien) [reply src/modules/stake-dot/StakeDotRoutes.tsx:19]: Yep, but it's not such trivial

**Pull request merged**